### ISSUE TITLE
Proposal: Decentralized Prediction Markets on Kaia Using Olas Mech

### ISSUE THREAD

**Issue by jorshimayor** [kaiachain/kaia-dapp-mono#78]
Description
- The platform will enable users to create markets, participate in trading, and settle outcomes in a trustless manner. Olas Mech's autonomous agent capabilities will be employed to automate market creation, liquidity provision, and result verification, enhancing the efficiency and reliability of the prediction markets.

Bounty track
- Other

**Comment by kaia-dev-intern** [comment]
this reads interesting

**Comment by zxstim** [comment]
I somewhat doubt the capability of autonomous agents because they hallucinate but you can take a crack at it and revert back here.

i think eigenlayer could also work for an optimistic oracle.

**Comment by jorshimayor** [comment]
Alright, I will. They do hallucinate, I'll try using a couple of models. 

I'll revert back, also is there any timeline to this?

**Comment by zxstim** [comment]
Nope. Take your time. i think a small POC is ok, keep the scope small so we all can assist you.

**Comment by jorshimayor** [comment]
Aight. Thanks a lot! I'll do that.

**Comment by jorshimayor** [comment]
Hey, I just wanted to ask if using other tools like ChainLink oracles is fine or is there any one we are to use specifically?

**Comment by zxstim** [comment]
kaia doesn't have chainlink. But use whatever necessary

**Comment by jorshimayor** [comment]
Hey, sorry it took this long, a lot of back and forth when I was building.

I was able to integrate UMA protocol on Sepolia testnet whilst interacting with it using an Oracle relayer from my Prediction Market smart contract deployed on Kaia.

Here's the GitHub repo: https://github.com/jorshimayor/Prediction-Market

Olas isn't deployed on Kaia when I asked the team and mostly supports Gnosis chain so I couldn't work on that.

I'm thinking on building out the frontend of this Issue pending your approval.

**Comment by zxstim** [comment]
hmm... why don't you merge this and your other proposal. i think the important piece in your proposal is having an optimistic oracle deployed on kaia

there are attempts to cross chain the events but native deployment is prob better in the long term

**Comment by jorshimayor** [comment]
Oh okay, I'll do that. 

I agree, native deployment is the best actually. Cross chain have been kinda hectic during development ngl.

I'll edit the forementioned issue now.

**Comment by zxstim** [comment]
If you can get the basic pieces of uma.xyz, i will do the first bounty payout immediately.

there is another alternative from https://truemarkets.org/

**Comment by zxstim** [comment]
since it will be a large community effort, join the convo here https://github.com/kaiachain/kaia-dapp-mono/discussions/236 and share your progress.

I will get the community to help you on things that you may need along the way

**Comment by jorshimayor** [comment]
> If you can get the basic pieces of uma.xyz, i will do the first bounty payout immediately.
> 
> there is another alternative from https://truemarkets.org/

Oh thanks for the alternative, I'll check that out now. 

For UMA, I've already integrated it to my smart contract, I use it to handle dispute after an event have being created on Kaia from my PredictionMarket smart contract. I also use Kaia to handle bets, payout and winning claims.

Thanks again for the support, I really appreciate it.

**Comment by jorshimayor** [comment]
> since it will be a large community effort, join the convo here #236 and share your progress.
> 
> I will get the community to help you on things that you may need along the way

Thanks alot, I'll do that now!

**Comment by zxstim** [comment]
I will check on how you handle the cross chain part in this proposal and assign a bounty for your work on that.

**Comment by jorshimayor** [comment]
Thanks very much, I really appreciate.

Alot of effort went into it. I'm also planning on working on a video format so we can post it on other channels.

Let me know what you think.
